scotty

```diff
@@ -0,0 +1,456 @@
+
+
+def install(args):
+    """ Install things
+    """
+
+    is_ras_pi = False
+
+    # if argument is passed and it's valid, set is_ras_pi if it's --pi
+    if len(args) > 0 and args[0] in ['--pi', '--dev']:
+        is_ras_pi = (args[0] == '--pi')
+
+    # Otherwise autodetect
+    else:
+        plat = get_platform()
+        if plat not in ['linux', 'raspberry-pi']:
+            print_color("\nInstalls on non linux systems are not supported yet! :(", color="red")
+            sys.exit()
+
+        is_ras_pi = (plat == 'raspberry-pi')
+
+    if is_ras_pi:
+        print_color("\nInstalling components for Raspberry Pi")
+    else:
+        print_color("\nInstalling components for dev system")
+
+    # Check if we have root access
+    if os.geteuid() != 0:
+        print_color("\nYou don't have permissions to install items. Exiting", color="red")
+        sys.exit()
+
+    install_items = []
+
+    # update and --fix-missing package manager
+    update_command = [
+        "apt-get update",
+        "apt-get update --fix-missing"
+    ]
+    run_commands(update_command)
+
+
+    # install basic things
+    install_items.extend(Config.get("aptget"))
+
+    # install ras pi things
+    if is_ras_pi:
+        install_items.extend(Config.get("aptget-hw"))
+
+    install_command = [
+        "apt-get install -y " + " ".join(install_items),
+        "pip install --upgrade pip",
+        "pip install --upgrade virtualenv"
+    ]
+
+    run_commands(install_command)
+
+
+    if not is_ras_pi:
+        print_color("\nInstalling and activating virtual env")
+        create_venv()
+        start_venv()
+
+    platform_specific_installs = []
+
+    platform_specific_installs.append(
+        "pip install -r %s" % get_file_path(Config.get("requirements"))
+    )
+
+    # install pi and dev specific things
+    if is_ras_pi:
+        platform_specific_installs.append(
+            "pip install -r %s" % get_file_path(Config.get("requirements-hw"))
+        )
+    else:
+        platform_specific_installs.append(
+            "pip install numpy"
+        )
+
+    run_commands(platform_specific_installs)
+
+    if os.environ['SUDO_USER'] and not is_ras_pi:
+        print_color("\nRunning with sudo, so chown the virtual env dir to the sudo user.")
+        command = "chown -R {user}:{user} {dir}".format(user=os.environ['SUDO_USER'],
+                                                        dir=get_file_path(Config.get("venv")))
+        run_commands([command])
+
+    if is_ras_pi or '--cam' in args:
+        if os.path.isdir(get_file_path('install/mjpg-streamer')):
+            print_color("\nmjpg-streamer folder already exists! Not installing.")
+        else:
+            install_cam = [
+                "apt-get install -y " + " ".join(Config.get("aptget-cam")),
+                "ln -sf /usr/include/linux/videodev2.h /usr/include/linux/videodev.h",
+                "git clone https://github.com/jacksonliam/mjpg-streamer.git install/mjpg-streamer",
```

Comment:
wait, this now clones the mjpg-streamer git. Before it used the one that's included in the `install` folder, AND is still included in this PR

Reply:
scotty always installed mjpg-streamer, it just did it from a localized repository. I believe I did that to trim down some of the files and because we knew this version worked.

Reply:
Yes...

It installs mjpg-streamer in the right spot if it doesn't already exist, in the event mjpg-streamer isn't installed prior in the repo. I mean shouldn't scotty install EVERYTHING needed to run the ROV? Why would mjpg-streamer be an exception?

Reply:
Okay, but the added code is minimal and I think its easier to have everything together in one install command.

Reply:
Okay, but it doesn't need this command: `git clone https://github.com/jacksonliam/mjpg-streamer.git install/mjpg-streamer` because the code it would be cloning is already in THIS repo

Reply:
So why would you reclone it?